File: src/main/java/com/heekng/celloct/dto/JoinRequestDto.java
package com.heekng.celloct.dto;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

public class JoinRequestDto {

    @Getter
    @Setter
    @NoArgsConstructor
    public static class JoinRequest {
        private Long memberId;
        private Long shopId;

        @Builder
        public JoinRequest(Long memberId, Long shopId) {
            this.memberId = memberId;
            this.shopId = shopId;
        }
    }

    @Getter
    @Setter
    @NoArgsConstructor
    public static class ManagerShopJoinRequestResponse {
        private Long joinRequestId;
        private Long memberId;
        private String memberName;
        private LocalDateTime createDate;

        @Builder
        public ManagerShopJoinRequestResponse(com.heekng.celloct.entity.JoinRequest joinRequest) {
            this.joinRequestId = joinRequest.getId();
            this.memberId = joinRequest.getMember().getId();
            this.memberName = joinRequest.getMember().getName();
            this.createDate = joinRequest.getCreateDate();
        }
    }

    @Getter
    @Setter
    @NoArgsConstructor
    public static class ApprovalRefusalRequest {
        private Long joinRequestId;

        @Builder
        public ApprovalRefusalRequest(Long joinRequestId) {
            this.joinRequestId = joinRequestId;
        }
    }

}

File: src/main/kotlin/com/heekng/celloct/dto/JoinRequestDto.kt
package com.heekng.celloct.dto

import com.heekng.celloct.entity.JoinRequest
import java.time.LocalDateTime

class JoinRequestDto {
    data class JoinRequest (
        var memberId: Long,
        val shopId: Long
    )

    data class ManagerShopJoinRequestResponse (
        val joinRequestId: Long?,
        val memberId: Long?,
        val memberName: String,
        val createDate: LocalDateTime?,
    ) {
        constructor(joinRequest: com.heekng.celloct.entity.JoinRequest): this(
            joinRequestId = joinRequest.id,
            memberId = joinRequest.member.id,
            memberName = joinRequest.member.name,
            createDate = joinRequest.createDate,
        )
    }

    data class ApprovalRefusalRequest (val joinRequestId: Long)
}